crates/polars-core/src/utils/mod.rs

```diff
@@ -34,6 +35,35 @@ impl<T> Deref for Wrap<T> {
     }
 }
 
+// This is giant trust me bro
+#[repr(transparent)]
+#[derive(Debug)]
+pub struct RegexWrap<T>(pub T);
+
+impl<T> AsRef<T> for RegexWrap<T> {
+    fn as_ref(&self) -> &T {
+        &self.0
+    }
+}
+
+impl<T> Default for RegexWrap<T> {
+    fn default() -> Self {
+        unreachable!()
+    }
+}
+
+impl<T> PartialEq for RegexWrap<T> {
```

Comment:
This one I am sure how, the optimizer will try to compare expressions that have a RegexWrap within them and may call this

Reply:
Done